### ISSUE TITLE
error: value of type '[NSDocument]' has no member 'compactMap'

### ISSUE THREAD

**Issue by SCdF** [xi-editor/xi-mac#339]
Apologies if this is my fault, but it doesn't look like master compiles right now and I think I've done everything right

```
$ git clone --recurse-submodules https://github.com/xi-editor/xi-mac

...

$ xcodebuild

...

/Users/scdf/Code/Other/xi-mac/Sources/XiEditor/AppDelegate.swift:456:37: error: value of type '[NSDocument]' has no member 'compactMap'
        return NSApplication.shared.orderedDocuments
               ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~

** BUILD FAILED **


The following build commands failed:
        CompileSwift normal x86_64
        CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(2 failures)
(no-nvm) scdf at SCdF in ~/Code/Other/xi-mac on master
$ rustup update stable
info: syncing channel updates for 'stable-x86_64-apple-darwin'

  stable-x86_64-apple-darwin unchanged - rustc 1.30.1 (1433507eb 2018-11-07)

(no-nvm) scdf at SCdF in ~/Code/Other/xi-mac on master
$ xcodebuild -version
Xcode 9.2
Build version 9C40b
```

**Comment by cmyr** [comment]
@SCdF I suspect this is a swift version difference? Can you try updating your Xcode to at least 9.4?

**Comment by SCdF** [comment]
Ahh, damn. I'm guessing if there is no compatible upgrade on the App Store I'm on the latest right?

I'm still on Sierra, and don't think I can upgrade any time soon. (high sierra sounded really unstable when it came out and I can't find an iso for it anymore, and Mojave drops acceptable font rendering for external non retina monitors, and I tried it for like 20 minutes before my eyes started bleeding).

**Comment by cmyr** [comment]
As a hack you could try just patching locally, replacing the line in question with `.map { $0 as! Document }` or something? Our support target is 'the last two macOS releases' but I don't see any reason to break over something this minor so if that fixes your situation we'd definitely accept a PR.

**Comment by SCdF** [comment]
@cmyr changing that caused the compilation to succeed. Thanks for the help! I created a PR linked above

**Comment by mmatoszko** [comment]
Can we close it?

**Comment by SCdF** [comment]
Yep, the linked PR was merged and solved the compilation problems in XCode 9.2.
